tests/mocked_plans/test_base_plan.py

```diff
@@ -0,0 +1,34 @@
+#  Copyright 2023 Canonical Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Test all sample plans."""
+from unittest.mock import patch
+
+import pytest
+
+from cou.commands import CLIargs
+from cou.steps.analyze import Analysis
+from cou.steps.plan import generate_plan
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.nova_compute.get_instance_count", return_value=0)
+async def test_base_plan(_, sample_plans):
+    """Testing the base plans."""
+    args = CLIargs("plan", auto_approve=True)
+    model, exp_plan = sample_plans["base.yaml"]
```

Comment:
non-blocking: this ties a very generic test to knowledge of the individual filenames. Could we consider changing the fixture to be parametrized and return one sample plan at a time instead?

Reply:
obviously this doesn't really matter right now since we only have one yaml

Reply:
I was thinking about using [Parametrizing fixtures](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#parametrizing-fixtures), however that will mean that your yaml file require working with this test configurations, e.g. CLIArgs and all hypervisors being empty. Like this, we could not create custom yaml file with let say only plan for one hypervisors (e.g. `CLIargs("hypervisors", machines=["0"], auto_approve=True)`) and on another hand like that all tests will need to work for each yaml file, since each test will run for each yaml file.

Reply:
Let's discuss this further in our sync. I'm merging this PR for now, since we do not have more tests and yaml files for now, so it doesn't make much sense to implement it now.

Reply:
reported #356